tests/unit/FirebaseStorage.spec.js

```diff
@@ -0,0 +1,361 @@
+          $task.$resume();
+          expect(mockTask.resume).toHaveBeenCalled();
+        });
+
+        it('$pause', function() {
+          var mockTask = new MockTask();
+          var ref = firebase.storage().ref('thing');
+          var file = 'file';
+          var $task = null;
+          var digestFn = $firebaseUtils.compile;
+          spyOn(ref, 'put').and.returnValue(mockTask);
+          spyOn(mockTask, 'pause')
+          $task = $firebaseStorage.utils._$put(ref, file, digestFn, $q);
+          $task.$pause();
+          expect(mockTask.pause).toHaveBeenCalled();
+        });
+
+        it('then', function() {
+          var mockTask = new MockTask();
+          var ref = firebase.storage().ref('thing');
+          var file = 'file';
+          var $task = null;
+          var digestFn = $firebaseUtils.compile;
+          spyOn(ref, 'put').and.returnValue(mockTask);
+          spyOn(mockTask, 'then')
+          $task = $firebaseStorage.utils._$put(ref, file, digestFn, $q);
+          $task.then();
+          expect(mockTask.then).toHaveBeenCalled();
+        });
+
+        it('catch', function() {
+          var mockTask = new MockTask();
+          var ref = firebase.storage().ref('thing');
+          var file = 'file';
+          var $task = null;
+          var digestFn = $firebaseUtils.compile;
+          spyOn(ref, 'put').and.returnValue(mockTask);
+          spyOn(mockTask, 'catch')
+          $task = $firebaseStorage.utils._$put(ref, file, digestFn, $q);
+          $task.catch();
+          expect(mockTask.catch).toHaveBeenCalled();
+        });
+
+      });
+
+      describe('_$getDownloadURL', function () {
+        it('should call a storage ref getDownloadURL', function (done) {
+          var ref = firebase.storage().ref('thing');
+          var testUrl = 'https://google.com/';
+          var storage = $firebaseStorage(ref);
+          var promise = $q(function (resolve, reject) {
+            resolve(testUrl);
+            reject(null);
+          });
+          var testPromise = null;
+          spyOn(ref, 'getDownloadURL').and.returnValue(promise);
+          testPromise = $firebaseStorage.utils._$getDownloadURL(ref, $q);
+          testPromise.then(function (resolvedUrl) {
+            expect(resolvedUrl).toEqual(testUrl)
+            done();
+          });
+          $rootScope.$apply();
+        });
+
+      });
+
+      describe('_$delete', function () {
+
+        it('should call a storage ref delete', function (done) {
+          var ref = firebase.storage().ref('thing');
+          var fakePromise = $q(function (resolve, reject) {
+            resolve(null);
+            reject(null);
+          });
+          var testPromise = null;
+          var deleted = false;
+          spyOn(ref, 'delete').and.returnValue(fakePromise);
+          testPromise = $firebaseStorage.utils._$delete(ref, $q);
+          testPromise.then(function () {
+            deleted = true;
+            expect(deleted).toEqual(true);
```

Comment:
I'm confused as to how this will ever not pass since you set `deleted` to `true` right above this. I don't think this test is actually testing what we want.